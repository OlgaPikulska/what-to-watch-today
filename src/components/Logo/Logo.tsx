import React from "react";
import Image from "next/image";
import Link from "next/link";

export const Logo: React.FC = () => {
	return (
		<Link href="/">
			<Image
				src="/header/logo_what_to_watch_today.jpg"
				width={100}
				height={100}
				alt="What to watch today logo"
			/>
		</Link>
	);
};
