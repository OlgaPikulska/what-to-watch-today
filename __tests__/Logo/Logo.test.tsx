import { Logo } from "@/components/Logo/Logo";
import { render } from "@testing-library/react";

describe("Logo Component", () => {
	it("should match the snapshot", () => {
		const { container } = render(<Logo />);
		expect(container).toMatchSnapshot();
	});

	it("should render a link to the homepage", () => {
		const { getByRole } = render(<Logo />);
		const linkElement = getByRole("link");
		expect(linkElement).toHaveAttribute("href", "/");
	});

	it("should render the logo image with the correct attributes", () => {
		const { getByAltText } = render(<Logo />);
		const imageElement = getByAltText("What to watch today logo");
		expect(imageElement).toHaveAttribute(
			"src",
			"/_next/image?url=%2Fheader%2Flogo_what_to_watch_today.jpg&w=256&q=75",
		);
		expect(imageElement).toHaveAttribute("width", "100");
		expect(imageElement).toHaveAttribute("height", "100");
	});
});
