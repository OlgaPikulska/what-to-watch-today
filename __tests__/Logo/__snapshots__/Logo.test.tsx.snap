// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Component should match the snapshot 1`] = `
<div>
  <a
    href="/"
  >
    <img
      alt="What to watch today logo"
      data-nimg="1"
      decoding="async"
      height="100"
      loading="lazy"
      src="/_next/image?url=%2Fheader%2Flogo_what_to_watch_today.jpg&w=256&q=75"
      srcset="/_next/image?url=%2Fheader%2Flogo_what_to_watch_today.jpg&w=128&q=75 1x, /_next/image?url=%2Fheader%2Flogo_what_to_watch_today.jpg&w=256&q=75 2x"
      style="color: transparent;"
      width="100"
    />
  </a>
</div>
`;
